common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1465,10 +1557,13 @@ public static class AppShuffleMergePartitionsInfo {
     private final int shuffleMergeId;
     private final Map<Integer, AppShufflePartitionInfo> shuffleMergePartitions;
 
+    private int[] reduceIds;
```

Comment:
```suggestion
    private final AtomicReferen<int[]> reduceIds = new AtomicReference<>(new int[0]);
```

and suitably modify the get and set.
It is not thread safe otherwise